internal/metrics/api.go

```diff
@@ -81,39 +85,51 @@ func (cl *APIClient) SendMetrics(metrics []APIMetric) error {
 	}
 	body := bytes.NewBuffer(content)
 
-	// For the moment we only support distribution metrics.
-	// Other metric types use the "series" endpoint, which takes an identical payload.
-	req, err := http.NewRequest("POST", cl.makeRoute("distribution_points"), body)
-	if err != nil {
-		return fmt.Errorf("Couldn't create send metrics request:%v", err)
+	if breaker == nil {
+		var st gobreaker.Settings
+		st.Name = "POST distribution_points"
+		// 5 failing requests, 60 secs in open state
+		breaker = gobreaker.NewCircuitBreaker(st)
 	}
-	req = req.WithContext(cl.context)
 
-	defer req.Body.Close()
+	_, err = breaker.Execute(func() (interface{}, error) {
```

Comment:
We apply retry logic, with incremental backoff in another function that calls this method: https://github.com/DataDog/datadog-lambda-go/blob/main/internal/metrics/processor.go#L130. It might make more sense to move the circuit breaker there, otherwise this will still add latency to the function when the circuit breaker is open.

Reply:
cc: @lhelman, in case you missed this comment. BTW, your last commit addressing my comments looks great, thanks for the quick turn around.

Reply:
Sure, I saw that one, let me see how to implement the circuit breaker there

Reply:
thanks for the prompt review.

I pushed a new version with this suggestion.

Reply:
moved to processor @DarcyRaynerDD

Reply:
Thanks for the quick update. I think this feature/code looks good. One last thing, I probably should have mentioned during my first review, could we add a unit test that triggers the breaker logic to the processor? I just want to validate that it doesn't have any unintended side effects. After that we should be good to merge.

Reply:
If this is urgent or related to an escalation, we could potentially merge without marking a release. Or you could deploy use your forked branch

Reply:
I've added a test for the opening case of the circuit breaker.

Do you prefer the circuit breaker to be passed as an interface, to make it easier to be tested?

About merging/releasing, I'm already using my branch (for this fix to resolve the related escalation), so I'm ok either way.

Reply:
I think the test you added is perfect.  Can you merge master again, I've updated our github actions workflow so it should run on your forked pr.

Reply:
I just merged main (and pushed)

Reply:
I just changed the trigger from "consecutive" to "total" count.

I saw in our production environment that the the timeouts are
not consecutive, so the circuit breaker will open when the total
amount of errors will be reached.

And, we still have the Interval setting to define how that "total" count
gets reset.